install-scripts/mac/skm-install-mac.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+GIT_SKM_REPO=https://github.com/splashkit/skm.git
+MARK_CHECK="\xE2\x9C\x94"
+MARK_CROSS="\xE2\x9C\x95"
+
+HOME_PATH=~
+INSTALL_PATH="${HOME_PATH}/.splashkit"
+
+function has_git() {
+	git --help 2>&1 > /dev/null
+	return $?
+}
+
+function install_git() {
+	echo "macOS developer tools are currently not installed which are required for installation skm."
+	read -p "Would you like to install them now? " -n 1 -r
+	echo ""
+	if [[ $REPLY =~ [Yy]$ ]]
+	then
+		xcode-select --install
+	else
+    	echo "macOS developer tools not installed, please run: \"xcode-select --install\" in the terminal and then reinstall."
+		exit 1
+	fi
+}
+
+echo "Checking for macOS developer tools..."
+has_git
+if [ $? -ne 0 ]; then
+	install_git
+	has_git
+	if [ $? -ne 0 ]; then
+		printf "\t$MARK_CROSS Failed to install macOS developer tools. Please ensure you have \"git\" installed and try again.\n"
+		exit 1
+	fi
+else
+	printf "\t$MARK_CHECK macOS developer tools appears to be installed\n"
+fi
+
+echo "Checking for an existing version of SplashKit..."
+if [ -d "${INSTALL_PATH}" ]; then
+	printf "\t$MARK_CROSS SplashKit is already installed!\n"
```

Comment:
**suggestion:** I am curious of the usage of `printf` instead of `echo` across this file. Since we are always appending newline character `\n`, would it be more succinct to use `echo` instead? 🤔 

Furthermore, for `printf`, it might be better to use variable as data as [suggested](https://github.com/koalaman/shellcheck/wiki/SC2059) by `shellcheck`.

Reply:
`printf` worked with Unicode with no arguments - but it looks like `echo -e` achieves the same thing.